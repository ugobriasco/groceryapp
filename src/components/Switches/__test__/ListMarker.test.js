import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import Icon from  'react-native-vector-icons/MaterialIcons';

import {ListMarker, styles} from '../';

describe('ListMarker', () => {

	it('renders correctly', () => {
	const tree = renderer.create(<ListMarker />).toJSON();
	expect(tree).toBeTruthy();

	});

	it ('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	describe('size stzling props', () => {
		it('should render an icon with the selected size', () => {
			const rendered = renderer.create(<ListMarker size='50'/>);
			expect(rendered).toMatchSnapshot();	
		});
		it('should render an icon with a given color if isChecked=true', () => {
			const rendered = renderer.create(<ListMarker isChecked={true} color='red'/>);
			expect(rendered).toMatchSnapshot();	
		});
	});
});