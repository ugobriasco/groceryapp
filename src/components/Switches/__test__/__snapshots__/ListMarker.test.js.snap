// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMarker size stzling props should render an icon with a given color if isChecked=true 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    backgroundColor="rgba(0,0,0,0)"
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "red",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î£Œ
  </Text>
</View>
`;

exports[`ListMarker size stzling props should render an icon with the selected size 1`] = `
<View
  style={undefined}
>
  <View />
</View>
`;
